### ISSUE TITLE
Failure loading XSD with regular expression facet (as generated by OSS Nokalva asn1xsd)

### ISSUE THREAD

**Issue by wangnick-work** [brunato/xmlschema#45]
The following XSD construct fails to load:

```
    <xsd:simpleType name="PrintableString">
	<xsd:restriction base="xsd:string">
	    <xsd:pattern value="[ &apos;\(\)\+-:=\?A-Za-z]*"/>
	</xsd:restriction>
    </xsd:simpleType>

```

The offending construct `[\+-:]` is valid XSD, noting that a [charRange](https://www.w3.org/TR/xmlschema11-2/#nt-charRange) is allowed to start with a [singleCharEsc](https://www.w3.org/TR/xmlschema11-2/#nt-singleChar). However, the code in codepoints.py:parse_character_group:99 does not foresee that `s[i] in r'|.^?*+{}()'` could start a character range.

Traceback is as follows:
```
  File "C:\Users\swangnic\AppData\Local\Programs\Python\Python36\lib\site-packages\xmlschema\validators\simple_types.py", line 840, in _parse
    facets[child.tag] = XsdPatternsFacet(base_type, child, self.schema)
  File "C:\Users\swangnic\AppData\Local\Programs\Python\Python36\lib\site-packages\xmlschema\validators\facets.py", line 327, in __init__
    regex = get_python_regex(value)
  File "C:\Users\swangnic\AppData\Local\Programs\Python\Python36\lib\site-packages\xmlschema\regex.py", line 249, in get_python_regex
    char_group, pos = parse_character_class(xml_regex, pos)
  File "C:\Users\swangnic\AppData\Local\Programs\Python\Python36\lib\site-packages\xmlschema\regex.py", line 210, in parse_character_class
    char_group = XsdRegexCharGroup(xml_regex[group_pos:pos])
  File "C:\Users\swangnic\AppData\Local\Programs\Python\Python36\lib\site-packages\xmlschema\regex.py", line 65, in __init__
    self.add(char)
  File "C:\Users\swangnic\AppData\Local\Programs\Python\Python36\lib\site-packages\xmlschema\regex.py", line 140, in add
    self.positive.update(part)
  File "C:\Users\swangnic\AppData\Local\Programs\Python\Python36\lib\site-packages\xmlschema\codepoints.py", line 287, in update
    for cp in iter_code_points(parse_character_group(values), reverse=True):
  File "C:\Users\swangnic\AppData\Local\Programs\Python\Python36\lib\site-packages\xmlschema\codepoints.py", line 134, in iter_code_points
    items = sorted(items, reverse=reverse, key=lambda x: x if isinstance(x, int) else x[1])
  File "C:\Users\swangnic\AppData\Local\Programs\Python\Python36\lib\site-packages\xmlschema\codepoints.py", line 98, in parse_character_group
    yield ord(start_char), ord(end_char)
TypeError: ord() expected string of length 1, but NoneType found
```

Kind regards,
Sebastian

**Comment by brunato** [comment]
Found the bug: it's the _start_char_ variable, that is a typo from previous code. This issue can be fixed substituting `start_char` with `char` at line 97:

https://github.com/brunato/xmlschema/blob/0d4e2188aa0e51db93a9de2d49532e39d18dac0f/xmlschema/codepoints.py#L97

I'll include the fix in the next version ASAP.

Thanks
Davide

**Comment by brunato** [comment]
The fix is available with version 0.9.23.

Best regards
Davide
